src/core/operators.pm

```diff
@@ -566,26 +566,29 @@ sub INDIRECT_NAME_LOOKUP($root, *@chunks) is raw {
     $thing;
 }
 
-sub REQUIRE_IMPORT($package-name, *@syms) {
-    my $package = CALLER::OUR::($package-name);
-    my $who     = $package.WHO;
-    unless $who.EXISTS-KEY('EXPORT') {
-        die "Trying to import symbols @syms.join(', ') from '$package-name', but it does not export anything";
-    }
-    $who := $who<EXPORT>.WHO<DEFAULT>.WHO;
+sub REQUIRE_IMPORT($compunit, *@syms,:$target-package) {
+    my $handle := $compunit.handle;
+    my $DEFAULT := $handle.export-package()<DEFAULT>.WHO;
+    my $GLOBALish := $handle.globalish-package.WHO;
     my @missing;
+    # Set the runtime values for compile time stub symbols
     for @syms {
-        unless $who.EXISTS-KEY($_) {
+        unless $DEFAULT.EXISTS-KEY($_) {
             @missing.push: $_;
             next;
         }
-        OUTER::CALLER::{$_} := $who{$_};
+        OUTER::CALLER::{$_} := $DEFAULT{$_};
     }
     if @missing {
-        X::Import::MissingSymbols.new(:from($package-name), :@missing).throw;
+        X::Import::MissingSymbols.new(:from($compunit.short-name), :@missing).throw;
     }
-    $package
+    CALLER::UNIT::GLOBALish::.merge-symbols($GLOBALish);
+
+    $target-package.defined ??
+        INDIRECT_NAME_LOOKUP($GLOBALish,$target-package) !!
```

Comment:
INDIRECT_NAME_LOOKUP replicates what was happening before (which is probably not correct but that's for another patch). If we don't have a package to lookup just return the file path.